src/apps/replication/lib/mutation_log.cpp

```diff
@@ -794,28 +776,28 @@ ::dsn::task_ptr mutation_log::append(
     //
     // start to write
     //
-    //err = write_pending_mutations();
-    // not batching
     if (_issued_write.expired()) {
         if (_batch_buffer_bytes == 0)
         {
+            // not batching
             err = write_pending_mutations();
         }
-        // batching
         else
         {
+            // batching
             if (static_cast<uint32_t>(_pending_write->size()) >= _batch_buffer_bytes)
             {
+                // batch full, write now
                 err = write_pending_mutations();
             }
             else
             {
-                // waiting for batch write later
+                // batch not full, wait for batch write later
+                // TODO(qinzuoyan): add timer triger
```

Comment:
there is no need for timer trigger as the private log is always fixed by shared log when the replica server restarts. We intentionally deleted the timer here - that's why you see some options in replication not used:)